### PR TITLE
Leverage str_contains/str_starts_with

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | N/A
| License       | MIT
| Doc PR        | N/A

I'd like to use `str_contains()` and `str_starts_with()` whenever possible. On PHP 8, this is a native function and for all earlier versions, we can maintain the most efficient way to perform those operations in the polyfill package. And apart from that, I find the new functions more intuitive than the `strpos()` expressions I'm replacing here.

All code changes in this PR were automated, see FriendsOfPHP/PHP-CS-Fixer#5754

Of course, this is more than just a CS change. If you don't feel comfortable merging this change into 4.4, I can easily redo the PR for 5.4.

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Templating/TemplateNameParser.php [hunk 1]
```diff
@@ -50,11 +50,11 @@ public function parse($name)
         // normalize name
         $name = preg_replace('#/{2,}#', '/', str_replace('\\', '/', $name));
 
-        if (false !== strpos($name, '..')) {
+        if (str_contains($name, '..')) {
             throw new \RuntimeException(sprintf('Template name "%s" contains invalid characters.', $name));
         }
 
-        if (!preg_match('/^(?:([^:]*):([^:]*):)?(.+)\.([^\.]+)\.([^\.]+)$/', $name, $matches) || 0 === strpos($name, '@')) {
+        if (!preg_match('/^(?:([^:]*):([^:]*):)?(.+)\.([^\.]+)\.([^\.]+)$/', $name, $matches) || str_starts_with($name, '@')) {
```
Comment (OskarStark) [inline src/Symfony/Bundle/FrameworkBundle/Templating/TemplateNameParser.php:57]: Not directly related and maybe micro optimisation, but wouldn't it be faster to do:
```suggestion
        if (str_starts_with($name, '@') || !preg_match('/^(?:([^:]*):([^:]*):)?(.+)\.([^\.]+)\.([^\.]+)$/', $name, $matches)) {
```

Comment (derrabus) [reply src/Symfony/Bundle/FrameworkBundle/Templating/TemplateNameParser.php:57]: Yes, probably. But I'd like to keep the change list straightforward and not include additional manual changes for now.

Comment (OskarStark) [reply src/Symfony/Bundle/FrameworkBundle/Templating/TemplateNameParser.php:57]: Yes definitely not part of this PR 👍

**Review by javiereguiluz — APPROVED** [review]
A big 👍 from me ... new code is so much more readable 😍

**Comment by fabpot** [comment]
Even if I appreciate the effort in modernizing the code base, I don't see any benefits of such a change. The code is not really different and it introduces a new polyfill as a dep (which might make the code slower).
Pondering the pros and the cons, I'm :-1: for such a change until we require 8.1+.

**Comment by noniagriconomie** [comment]
@fabpot hi,

> The code is not really different and it introduces a new polyfill as a dep

according to https://packagist.org/packages/symfony/polyfill-php80/dependents?order_by=downloads, this "internal" package is required on major symfony packages already

> (which might make the code slower)

this indeed can be the main reason, when running on sf v4 or v5 without php v8n thus leveraging the polyfill, but I can not tell if it has an impact or not :s 

> until we require 8.1+.

I do not understand the `8.1+`, sf v6 which require php v8.0+ could be a good candidate no?

thx

**Comment by fabpot** [comment]
> > until we require 8.1+.
> 
> I do not understand the `8.1+`, sf v6 which require php v8.0+ could be a good candidate no?
> 

That's a typo, I meant 8.0

**Comment by Tobion** [comment]
@fabpot as already said by @noniagriconomie, the php80 polyfill is already a requirement in 4.4 by major symfony components like httpkernel and console. So it's already part of every symfony app. So there is no additional dependency.

I'd also argue it really makes the code mroe readable and it clarifies the intent. There are several workarounds implementations, some better suited than others (see rfc). By using the standard functions it also normalizes those inconsistencies.

You said it might make the code slower. I'd say it makes it faster, especially on PHP 8.0+. One of the arguments for introducing the functions was that the alternatives are not efficient.
Another argument is that SF 4.4 will be maintained a year longer than PHP 7.4. So at one point symfony 4.4 users are likely to run on PHP 8. Then they will get the performance increase automatically.

So to me there is no downside here and it makes the code future-proof. Please reconsider.

**Comment by fabpot** [comment]
ok, let's do it then, but we need to enable the rule on PHP CS fixer to be sure we will catch "old" usages in PRs.

**Comment by derrabus** [comment]
We need to get the rule merged into CS Fixer then. 😓

**Comment by nicolas-grekas** [comment]
Thank you @derrabus.

File: src/Symfony/Component/Translation/MessageCatalogue.php [hunk 2]
```diff
@@ -166,7 +166,7 @@ public function add($messages, $domain = 'messages')
         }
         $intlDomain = $domain;
         $suffixLength = \strlen(self::INTL_DOMAIN_SUFFIX);
-        if (\strlen($domain) < $suffixLength || false === strpos($domain, self::INTL_DOMAIN_SUFFIX, -$suffixLength)) {
+        if (\strlen($domain) < $suffixLength || !str_contains($domain, self::INTL_DOMAIN_SUFFIX, -$suffixLength)) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Translation/MessageCatalogue.php:169]: this change is wrong, it might be good to report it to php-cs-fixer

**Comment by nicolas-grekas** [comment]
Linking to https://github.com/composer/composer/issues/10024 for references.

**Pull request merged**